data/1.1/_index.md

```diff
@@ -0,0 +1,79 @@
+<!-- Template for the root page of a specification release -->
+<!-- The "date:" field is used as a "publish" date for the automated Hugo processing -->
+---
+title: "Jakarta Data 1.1 (under development)"
+date: 2025-01-28
+summary: "Enhancements to the Jakarta Data spec"
+---
+
+### New features, enhancements or additions
+* Ability to define query restrictions dynamically (https://github.com/jakartaee/data/issues/829)
+* Life cycle events (https://github.com/jakartaee/data/issues/373)
+* Bring parameter based automatic query methods up to the level of what Query by Method Name can do for static queries (https://github.com/jakartaee/data/issues/857)
+* Query Fragments (https://github.com/jakartaee/data/issues/546)
+* Multiple item select lists/DTOs/Entity Views (https://github.com/jakartaee/data/issues/539)
+* Consider additional annotations related to Jakarta Persistence (https://github.com/jakartaee/data/issues/470)
+* Reactive patterns (#17) and integration with Jakarta Concurrency Asynchronous (https://github.com/jakartaee/data/issues/19)
+* Move Jakarta Data Query Language to Jakarta Query specification (if accepted and included in Jakarta EE 12) where it will remain fully compatible.
+* Configuration via Jakarta Config (if Jakarta Config is released and makes it into Jakarta EE 12 in time)
+* Explicitly state where NoSQL databases can raise errors for function the database type is not capable of (https://github.com/jakartaee/data/issues/782)
+* TCK refactoring and enhancements to more logically split out tests that are targeted at various levels of database support (key-value, column, document, graph, relational).
+
+
+### Removals, deprecations or backwards incompatible changes
+* N/A
+
+### Minimum Java SE Version
+<!-- Specify the minimum required Java SE version for this specification -->
+**21**
```

Comment:
I don't think you can drop Java 17 in a minor release. What Java 21 feature did you use? If not, you can just update this. Otherwise, you need to remove that issues from the list.

Reply:
> I don't think you can drop Java 17 in a minor release. What Java 21 feature did you use? If not, you can just update this. Otherwise, you need to remove that issues from the list.

It is perfectly fine to do that in a minor release. This has been discussed over and over again in the platform call and always concluded with that.

Reply:
I am not aware of any Java 21-specific feature used.
I wrote 21 here because that is the minimum required Java SE level for Jakarta EE 12.  At this point we have no guarantee that all of the Jakarta EE 12 specifications that we depend on will opt to continue supporting Java SE 17.

Reply:
I disagree with dropping SE 17 without any particular reason. It is irrelevant whether Jakarta EE 12 has the minimum version of SE 21 or not. The individual specs need to specify which SE do they support. As long as one spec has the minimum Java SE version to be 21, EE 12 has to be SE 21 based.

Reply:
> I disagree with dropping SE 17 without any particular reason. It is irrelevant whether Jakarta EE 12 has the minimum version of SE 21 or not. The individual specs need to specify which SE do they support. As long as one spec has the minimum Java SE version to be 21, EE 12 has to be SE 21 based.

If you disagree, you can vote -1 on the ballot. But remember it is not a perquisite for starting the ballot that the mentor agrees with everything in the plan. ust check that the materials are okay and has the quality it needs for a review to start.

Reply:
I think there's a non-negligable impact on the TCK, @Emily-Jiang, if we keep Java 17. We would have to support Java 17, 21, and I suppose 25, if Jakarta EE 12 is aligned with 25. That makes one more Java version to support by the Data TCK. Maybe OK, maybe just unnecessary pain.

Reply:
On the Jakarta EE platform call, we confirmed that dependencies requiring a minimum Java SE 21 is a valid reason for a specification to also require a minimum of Java SE 21.  Looking through the features listed in the Data 1.1 plan, it includes interoperability with `Asynchronous` from Jakarta Concurrency (which will be minimum Java SE 21) which the Data TCK will need to run against.  My best guess is that a newly introduced Jakarta Query (which Data will depend on) would also have a minimum of Java SE 21.  It is difficult to imagine a newly created spec deciding to support Java SE 17 that will be going out of service so soon after.